pkg/k8s/statefulsets.go

```diff
@@ -42,11 +45,11 @@ func WatchStatefulSets() error {
 
 	// Watch for events on the watcher channel
 	for event := range watcher.ResultChan() {
+		// Check if the event object is of type *v1.StatefulSet
 		if statefulSet, ok := event.Object.(*v1.StatefulSet); ok {
-			//log.Info("StatefulSet containers: ", statefulSet.Spec.Template.Spec.Containers)
-
-			// check if the node is DA, if so, send it to the queue to generate the multi address
-			if strings.HasPrefix(statefulSet.Name, "da") {
+			// Check if the StatefulSet is valid based on the conditions
+			if isStatefulSetValid(statefulSet) {
+				// Perform necessary actions, such as adding the node to the Redis queue
```

Comment:
design wise, you can do the following: 

1. Check if not ok, fail fast with error and exit the for loop
2. If ok, then you have a second if outside of the boundaries of the first if else one(where you have the ok checker) 
3. If you want to check all the objects in the channel, you need to store the error messages one by one and then exit the for loop. So far, as I understood the code, you are not intending to do so